tests/ui_tests/app/page/admin/page.py

```diff
@@ -32,85 +34,124 @@
+
+
+@dataclass
+class AdminRoleInfo:
+    """Information about role"""
+
+    name: str
+    description: str
+    permissions: str
 
 
 class GeneralAdminPage(BasePageObject):
     """Base class for admin pages"""
 
+    MENU_SUFFIX: str
     MAIN_ELEMENTS: List[Locator]
+    header: PageHeader
+    footer: PageFooter
+    config: CommonConfigMenuObj
+    table: CommonTableObj
+    toolbar: CommonToolbar
+
+    def __init__(self, driver, base_url):
+        if self.MENU_SUFFIX is None:
+            raise AttributeError('You should explicitly set MENU_SUFFIX in class definition')
+        super().__init__(driver, base_url, "/admin/" + self.MENU_SUFFIX)
+        self.header = PageHeader(self.driver, self.base_url)
+        self.footer = PageFooter(self.driver, self.base_url)
+        self.config = CommonConfigMenuObj(self.driver, self.base_url)
+        self.table = CommonTableObj(self.driver, self.base_url)
+        self.toolbar = CommonToolbar(self.driver, self.base_url)
 
     @allure.step("Assert that all main elements are presented on the page")
     def check_all_elements(self):
         """Assert presence of the MAIN_ELEMENTS"""
+
         if len(self.MAIN_ELEMENTS) == 0:
             raise AttributeError('MAIN_ELEMENTS should contain at least 1 element')
         self.assert_displayed_elements(self.MAIN_ELEMENTS)
 
+    def check_admin_toolbar(self):
+        self.assert_displayed_elements([CommonToolbarLocators.admin_link])
+
     @allure.step('Open Admin Intro page by left menu item click')
-    def open_intro_menu(self):
+    def open_intro_menu(self) -> "AdminIntroPage":
         """Open Admin Intro page by menu object click"""
+
         self.find_and_click(ObjectPageMenuLocators.intro_tab)
+        page = AdminIntroPage(self.driver, self.base_url)
+        page.wait_page_is_opened()
+        return page
 
     @allure.step('Open Admin Settings page by left menu item click')
-    def open_settings_menu(self):
+    def open_settings_menu(self) -> "AdminSettingsPage":
         """Open Admin Settings page by menu object click"""
+
         self.find_and_click(ObjectPageMenuLocators.settings_tab)
+        page = AdminSettingsPage(self.driver, self.base_url)
+        page.wait_page_is_opened()
+        return page
 
     @allure.step('Open Admin Users page by left menu item click')
-    def open_users_menu(self):
+    def open_users_menu(self) -> "AdminUsersPage":
         """Open Admin Users page by menu object click"""
+
         self.find_and_click(ObjectPageMenuLocators.users_tab)
+        page = AdminUsersPage(self.driver, self.base_url)
+        page.wait_page_is_opened()
+        return page
+
+    @allure.step('Open Admin Roles page by left menu item click')
+    def open_roles_menu(self) -> "AdminRolesPage":
+        """Open Admin Roles page by menu object click"""
+
+        self.find_and_click(ObjectPageMenuLocators.roles_tab)
+        self.find_and_click(ObjectPageMenuLocators.roles_tab)
```

Comment:
Why do we need to click it twice? Please add comment

Reply:
yep, removed